src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Operations/Reindex/ReindexProcessingJobTests.cs

```diff
@@ -260,5 +243,665 @@ public async Task GivenSurrogateIdRange_WhenHashDoesNotMatch_ThenRaiseError()
+                GroupId = 1,
+                CreateDate = DateTime.UtcNow,
+                Status = JobStatus.Running,
+            };
+
+            var searchResultEntries = Enumerable.Range(1, 2)
+                .Select(i => CreateSearchResultEntry(i.ToString(), expectedResourceType))
+                .ToList();
+
+            _searchService.SearchForReindexAsync(
+                Arg.Any<IReadOnlyList<Tuple<string, string>>>(),
+                Arg.Any<string>(),
+                false,
+                Arg.Any<CancellationToken>(),
+                true)
+                .Returns(new SearchResult(
+                    searchResultEntries,
+                    null,
+                    null,
+                    new List<Tuple<string, string>>()));
+
+            var result = await _reindexProcessingJobTaskFactory().ExecuteAsync(jobInfo, _cancellationToken);
+
+            Assert.NotEmpty(result);
+            var jobResult = JsonConvert.DeserializeObject<ReindexProcessingJobResult>(result);
+            Assert.NotNull(jobResult);
+            Assert.Equal(2, jobResult.SucceededResourceCount);
+        }
+
+        [Fact]
+        public async Task ValidateSearchParametersHash_WithEmptyCurrentHash_ThrowsException()
+        {
+            var expectedResourceType = "Observation";
+            var job = new ReindexProcessingJobDefinition()
+            {
+                MaximumNumberOfResourcesPerQuery = 100,
+                MaximumNumberOfResourcesPerWrite = 100,
+                ResourceType = expectedResourceType,
+                ResourceCount = new SearchResultReindex()
+                {
+                    Count = 1,
+                    EndResourceSurrogateId = 100,
+                    StartResourceSurrogateId = 1,
+                },
+                ResourceTypeSearchParameterHashMap = "observationHash",
+                SearchParameterUrls = new List<string>() { "http://hl7.org/fhir/SearchParam/Observation-code" },
+                TypeId = (int)JobType.ReindexProcessing,
+            };
+
+            // Return empty hash to trigger validation error
+            _searchParameterOperations.GetResourceTypeSearchParameterHashMap(Arg.Any<string>()).Returns(string.Empty);
+
+            var jobInfo = new JobInfo()
+            {
+                Id = 1,
+                Definition = JsonConvert.SerializeObject(job),
+                QueueType = (byte)QueueType.Reindex,
+                GroupId = 1,
+                CreateDate = DateTime.UtcNow,
+                Status = JobStatus.Running,
+            };
+
+            var exception = await Assert.ThrowsAsync<ReindexProcessingJobSoftException>(
+                () => _reindexProcessingJobTaskFactory().ExecuteAsync(jobInfo, _cancellationToken));
+
+            Assert.Contains("hash does not match", exception.Message);
+        }
+
+        [Fact]
+        public async Task ValidateSearchParametersHash_WithNullCurrentHash_ThrowsException()
```

Comment:
Nit: could this test and the one above it be a Theory? It looks like the only difference is the hash value.